### DISCUSSION TITLE
翻訳プラットフォームの導入

### DISCUSSION THREAD

**Discussion by L4Ph** [Katsuyuki-Karasawa/stable-diffusion-webui-localization-ja_JP#3] [Ideas]
ファイルを直接いじるよりは差分をとってGUIで変更が楽
現在想定しているところでは、
1. GItLocalize
    - GItHubと密接なのでいまのところ一番よさそう
    - https://gitlocalize.com/repo/8200
      - 設置してみました
2. Crowdin 
    - あまりいい思い出がない(リポジトリが分散してなければあり?)
    - crowdin-jsonだと勝手に出力される形式が変わるのでよろしくない挙動をする
    - i18next jsonだと問題ないように見える?
3. Transifex
    - https://github.com/Katsuyuki-Karasawa/stable-diffusion-webui-localization-ja_JP/discussions/3#discussioncomment-4242236
    - とはいえ、どのプラットフォームでも似たような状況になりそうなので、使用感によってはあり?

候補が出てきたら追加する(デモや実際に試した所感)

**Comment by krisfail** [comment]
大いに賛成です。個人的には同じ単語に同じ訳を当てる機能があるCrowdinを勧めたいです。(GitLocalizeにはなさそう?)

**Reply by L4Ph** [reply]
crowdinにレビュー機能がなかったような気がする(ブランチが別で生成されるのでmainにpushするときに修正すればいいだけではある)のですがどうなんでしょう?

**Reply by krisfail** [reply]
翻訳側で参加したことがあるだけなんでなんとも言えませんね…
まあお任せします

**Comment by mkpoli** [comment]
Transifex は認証取ればオープンソース向けに無料ですが、普通は翻訳元（WebUI の作者ら）の人が組織するので、この場合だと微妙かもですが

**Comment by L4Ph** [comment]
自動で`crowdin_main`をマージ後に削除するように設定変更しました
挙動を見てると､どうやら`main`にマージしたあとに`crowdin_main`を削除→`main`を追従して`crowdin_main`を生成のようだったので､安心してブランチ削除していただいて大丈夫です

**Reply by L4Ph** [reply]
少し気になったので実験してみます(原文からコピーした翻訳は`crowdin_main`を生成したときにリセットされる...?のであまり意味がないかもしれない気がしてきました)

**Reply by L4Ph** [reply]
確認してみましたが､そんなことはないみたいですね...
どこかでconflictしたタイミングで翻訳lostしたかな...

**Reply by harukaxxxx** [reply]
こちの経験は全ての翻訳はCrowdinで管理する、なぜなら、crowdin_mainの翻訳済みのファイルは毎回リセットされる。
そして、全ての英語のtemplate更新は、mainだけで更新する、Crowdinは自動で更新しまる。
だから、crowdinではなくの翻訳作業は(jsonファイルでいじる)、Crowdinからアップロードでマージするのはいいと思うです。
